packages/nuxt/src/core/plugins/resolve-deep-imports.ts

```diff
@@ -0,0 +1,25 @@
+import { parseNodeModulePath, resolvePath } from 'mlly'
+import { isAbsolute } from 'pathe'
+import type { Plugin } from 'vite'
+import type { Nuxt } from '@nuxt/schema'
+import { pkgDir } from '../../dirs'
+
+export function resolveDeepImportsPlugin (nuxt: Nuxt): Plugin {
+  return {
+    name: 'nuxt:resolve-bare-imports',
+    enforce: 'post',
+    async resolveId (id, importer, options) {
+      if (!importer || isAbsolute(id) || !isAbsolute(importer) || id.startsWith('virtual:')) {
+        return
+      }
+      const { dir } = parseNodeModulePath(importer)
+      return await this.resolve?.(id, dir || pkgDir, { skipSelf: true }) ?? await resolvePath(id, {
+        url: [dir || pkgDir, ...nuxt.options.modulesDir],
+        conditions: options.ssr ? ['node', 'import', 'require'] : ['import', 'require']
```

Comment:
Note: With nitro, we use runtime environment condition sets. Not sure if we can easily get resolve conditions from nitro now but a TODO would be nice to consider